configextractor_/update_server.py

```diff
@@ -54,46 +60,66 @@ def import_parsers(cx: ConfigExtractor):
                             )
                         ).as_primitives()
                     )
-            return (
-                client.signature.add_update_many(source_name, "configextractor", upload_list, dedup_name=False)
-            )
+            return client.signature.add_update_many(source_name, "configextractor", upload_list, dedup_name=False)
 
         for dir, _ in files_sha256:
             # Remove cached duplicates
             dir = dir[:-1]
             self.log.info(dir)
 
-            PYTHON_PACKAGE_EXCL = ['yara-python', 'maco', 'pefile']
+            PYTHON_PACKAGE_EXCL = ["yara-python", "maco"]
 
             # Find any requirement files and pip install to a specific directory that will get transferred to services
             # Limit search for requirements.txt to root of folder containing parsers
             if "requirements.txt" in os.listdir(dir):
+                req_path = os.path.join(dir, "requirements.txt")
+                mod_req_path = os.path.join(dir, "modified_requirements.txt")
+                with open(req_path, "r") as req_fp:
+                    req = [
+                        r.strip() for r in req_fp.readlines() if not any([r.startswith(x) for x in PYTHON_PACKAGE_EXCL])
+                    ]
+                    req_pkgs = [r.split("==")[0] for r in req]
+                    # Get the package dependencies for each required dependency
+                    deps = []
+                    for r in req:
+                        dist = JohnnyDist(r, ignore_errors=True)
+                        [
+                            (deps.append(d), req_pkgs.append(d.split("==")[0]))
+                            for d in dist.serialise(format="pinned").split("\n")
+                            if d.split("==")[0] not in req_pkgs  # Prevent package version conflicts with requirements
+                            and not d.split("==")[0] in PYTHON_PACKAGE_EXCL  # Prevent package conflicts with container
+                            and d not in deps  # Prevent package version conflicts in the dependency list
```

Comment:
I don't think you can get to this line and it being False.
If d is in deps, it needs to be in req_pkgs as the first part, which will be caught in line 89 to skip to the next d.